cluster.go

```diff
@@ -16,11 +16,13 @@ type ClusterClient struct {
 
 	addrs map[string]struct{}
 	slots [][]string
+	sLock sync.RWMutex // protects slots & addrs cache
```

Comment:
Can you please use full name, e.g. `slotsLock`? Difference between `sLock` and `cLock` is too subtle.

Also the usual name is `slotsMutex` or `slotsMtx` to avoid double "lock" word in `theLock.Lock`.